hypershift-operator/controlplaneoperator-overrides/overrides.go

```diff
@@ -0,0 +1,84 @@
+package controlplaneoperatoroverrides
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+
+	"github.com/openshift/hypershift/support/config"
+	"gopkg.in/yaml.v2"
+)
+
+type CPOOverrides struct {
+	Overrides []CPOOverride `yaml:"overrides,omitempty"`
+}
+
+type CPOOverride struct {
+	Version  string                  `yaml:"version"`
+	CPOImage string                  `yaml:"cpoImage"`
+	Testing  CPOOverrideTestReleases `yaml:"testing"`
+}
+
+type CPOOverrideTestReleases struct {
+	Latest   string `yaml:"latest"`
+	Previous string `yaml:"previous"`
+}
+
+//go:embed assets/overrides.yaml
+var overridesYAML []byte
+
+var (
+	overrides          = mustLoadOverrides()
+	overridesByVersion map[string]*CPOOverride
+)
+
+func init() {
+	initOverridesByVersion()
+}
+
+func IsOverridesEnabled() bool {
+	return os.Getenv(config.CPOOverridesEnvVar) == "1"
+}
+
+func CPOImage(version string) string {
+	return lookupCPOImage(version)
+}
+
+func LatestOverrideTestReleases() (string, string) {
+	if len(overrides.Overrides) > 0 {
+		testingReleases := overrides.Overrides[len(overrides.Overrides)-1].Testing
+		return testingReleases.Latest, testingReleases.Previous
```

Comment:
This implies that the overrideReleases should follow a concrete order in the file?

Reply:
Only for testing. The thinking is that when we test overrides, we only test the latest mapping.

Reply:
why do we need to specify `testing` field for each version then?

Reply:
because just the version is not enough to know which release image to use

Reply:
hmm, I mean if all other testing entries are not used because always use the last entry, why not have a single testing field? I 
```go
type CPOOverrides struct {
	Overrides []CPOOverride `yaml:"overrides,omitempty"`
	Testing CPOOverrideTestReleases `yaml:"testing"`
}
```

Reply:
Makes sense, updated